lib/govuk_app_config/govuk_prometheus_exporter.rb

```diff
@@ -1,14 +1,34 @@
 module GovukPrometheusExporter
   def self.configure
-    unless Rails.env == "test" || (ENV["GOVUK_PROMETHEUS_EXPORTER"]) != "true"
+    if (ENV["GOVUK_PROMETHEUS_EXPORTER"]) == "true"
+      return if defined?(Rails) ? Rails.env == "test" : false
+
       require "prometheus_exporter"
       require "prometheus_exporter/server"
       require "prometheus_exporter/middleware"
 
+      if defined?(Sidekiq)
+        Sidekiq.configure_server do |config|
+          require "prometheus_exporter/instrumentation"
+          config.server_middleware do |chain|
+            chain.add PrometheusExporter::Instrumentation::Sidekiq
+          end
+          config.death_handlers << PrometheusExporter::Instrumentation::Sidekiq.death_handler
+          config.on :startup do
+            PrometheusExporter::Instrumentation::Process.start type: "sidekiq"
+            PrometheusExporter::Instrumentation::SidekiqProcess.start
+            PrometheusExporter::Instrumentation::SidekiqQueue.start
+            PrometheusExporter::Instrumentation::SidekiqStats.start
+          end
+        end
+      end
+
       server = PrometheusExporter::Server::WebServer.new bind: "0.0.0.0", port: 9394
       server.start
 
-      Rails.application.middleware.unshift PrometheusExporter::Middleware
+      if defined?(Rails)
+        Rails.application.middleware.unshift PrometheusExporter::Middleware
```

Comment:
Is it preferable to use the [`config.middleware` interface](https://guides.rubyonrails.org/configuring.html#adding-custom-middleware) here rather than manipulating `application.middleware` directly?
```suggestion
        Rails.application.config.middleware.insert_before 0, PrometheusExporter::Middleware
```

Reply:
I copied it across from the documentatio: https://github.com/discourse/prometheus_exporter/blob/main/README.md#rails-integration